hugo/content/guides/building_an_extension.md

```diff
@@ -0,0 +1,47 @@
+
+## Setting up the Scripts
+
+To get started, you'll want to have a language expressed in Langium, such as [Lox](https://github.com/langium/langium-lox) or [MiniLogo](https://github.com/langium/langium-minilogo). If you have been following along with these guides, you should already have something ready. If you don't you can also use the default language generated by the yeoman generator for Langium, presented in the [getting started](/docs/getting-started/) section.
+
+Regardless of what you're working with, you'll want to make sure you have the following scripts in your **package.json**.
+
+```json
+{
+    ...,
+    "vscode:prepublish": "npm run esbuild-base -- --minify && npm run lint",
+    "esbuild-base": "esbuild ./src/extension.ts --bundle --outfile=out/main.js --external:vscode --format=cjs --platform=node",
+    "lint": "eslint src --ext ts"
+}
+```
+
+The `esbuild-base` script is particularly important, as it will be constructing the extension itself.
+
+You'll also need to install `esbuild` if you haven't already.
+
+```bash
+npm i --save-dev esbuild
+```
+
+At this point we're ready to generate an extension. Ws need the VS Code Extension Manager (`vsce`) to do this, so make sure to download this from npm via `npm install -g vsce` (or install locally, as per your preference). Once you have that installed, you can invoke it like so from the root of your project.
+
+```bash
+vsce package
+```
+
+You should now see a VSIX extension file in the root of your project. The name of this file will correspond with the **name** and **version** properties listed in your **package.json**. For MiniLogo, this produced **minilogo-0.1.0.vsix**.
+
+For installing the extension, you can right click the extension file, and select "Install VSIX Extension" at the bottom of the list. You should see a small indication that your VSIX extension has been successfully installed. You can verify this by going to your extensions tab and looking at the enabled extensions, where you should find the name of your language (again corresponding to the **name** property in your package.json). Assuming the extension is enabled and working correctly, you can open any file that ends in the extensions registered for your language, and you should immediately observe the syntax highlighting kicking in. Interaction with your language should show that syntax errors are recognized, and other LSP functionalities are working as intended (such as renaming of symbols).
+
+You may also notice that your extension may not have an icon to start with. This is a small thing that we can fix quickly. This is as simple as adding a small PNG icon somewhere in your project repo, such as the root. You'll also want to set the **icon** property in your package.json with the relative path to this icon. When you regenerate your extension & reinstall it, you should get an icon that is the same as the one that you packaged it with.
```

Comment:
Add an icon? That is nice and may deserves an own headline?
These points can be also part of an agenda at the beginning of the entire text

Reply:
That can be added. I wonder if geekdoc supports an auto-ToC feature, could help with organization at the start.

Reply:
This has it's own sub-header now. But the local TOC I have to look again. Some hugo themes will automatically inject the table of contents at the start, others need a flag to do so, and some don't have it

Reply:
Local TOC is also setup now 👍